sw/device/lib/testing/test_main.c

```diff
@@ -7,9 +7,14 @@
 #include "sw/device/lib/arch/device.h"
 #include "sw/device/lib/base/log.h"
 #include "sw/device/lib/base/print.h"
+#include "sw/device/lib/testing/test_coverage.h"
 #include "sw/device/lib/testing/test_status.h"
 #include "sw/device/lib/uart.h"
 
+// This weak definition is overridden by the actual definition in
+// `test_coverage.c` when coverage is enabled.
+__attribute__((weak)) void test_coverage_send_buffer(void) {}
```

Comment:
We're trying to avoid `__attribute__((weak))`.

I think the solution here is to have two "implementations" of `test_coverage.h` - `test_coverage_llvm.c` and `test_coverage_none.c` - the second of which just has an empty function here. Thus we can always link in `test_coverage_none.o`, unless we have coverage enabled where we'd link in `test_coverage_llvm.o`. Does that make sense?

Reply:
It definitely does. Apologies if we've already talked about this and I forgot about it. Done, PTAL.